src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -41,16 +41,29 @@ internal static partial class PathInternal
         /// </summary>
         internal static bool IsPathTooLong(string fullPath)
         {
+            // We'll never know precisely what will fail as paths get changed internally in Windows and
+            // may grow to exceed MaxExtendedPath. We'll only try to catch ones we know will absolutely
+            // fail.
+
+            if (fullPath.Length < MaxExtendedPath - UncExtendedPathPrefix.Length)
+            {
+                // For sure is ok
```

Comment:
Why?  The previous comment just stated that paths can grow, and it's not clear why the checked condition makes this ok.  Please augment the comment to explain fully.